test/e2e/federated-namespace.go

```
@@ -0,0 +1,118 @@
+				if _, found := clusterClientSet[cluster.Name]; !found {
+					clientset, err := util.GetClientsetForCluster(cluster)
+					if err != nil {
+						framework.Failf("Failed to create client for %s: %v", cluster.Name, err)
+					}
+					clusterClientSet[cluster.Name] = clientset
+				}
+			}
+		})
+
+		AfterEach(func() {
+			framework.SkipUnlessFederated(f.Client)
+			deleteAllTestNamespaces(f.FederationClientset_1_4)
+			for _, clientset := range clusterClientSet {
+				deleteAllTestNamespaces(clientset)
+			}
+		})
+
+		It("should be created and deleted successfully", func() {
+			framework.SkipUnlessFederated(f.Client)
+
+			ns := api_v1.Namespace{
+				ObjectMeta: api_v1.ObjectMeta{
+					Name: fmt.Sprintf("%s%d", namespacePrefix, time.Now().UnixNano()),
+				},
+			}
+			By(fmt.Sprintf("Creating namespace %s", ns.Name))
+			_, err := f.FederationClientset_1_4.Core().Namespaces().Create(&ns)
+			framework.ExpectNoError(err, "Failed to create namespace %s", ns.Name)
+
+			// Check subclusters if the namespace was create there.
+			err = wait.Poll(5*time.Second, 2*time.Minute, func() (bool, error) {
+				for _, client := range clusterClientSet {
+					_, err := client.Core().Namespaces().Get(ns.Name)
+					if err != nil {
+						return false, err
+					}
+					// TODO: Uncomment once the namespace controller is fully enabled.
+					// if item == nil {
+					// 	return false, nil
+					// }
+				}
+				return true, nil
+			})
+			// This is a dummy check, the above snippent must be enabled.
+			framework.ExpectNoError(err, "Not all namespaces created")
+
+			err = f.FederationClientset_1_4.Core().Namespaces().Delete(ns.Name, &api.DeleteOptions{})
+			framework.ExpectNoError(err, "Failed to delete namespace %s", ns.Name)
+		})
+
+	})
+})
+
+func deleteAllTestNamespaces(clientset *federation_release_1_4.Clientset) {
+	list, err := clientset.Core().Namespaces().List(api.ListOptions{})
+	if err != nil {
+		framework.Logf("Failed to get all namespaes: %v", err)
```

The let's change the function to return an error and return an error here.

Currently you are silently dropping errors, which is bad.